### DISCUSSION TITLE
CoMD Analysis Issue - Coordsets must have more than 3 coordinate sets

### DISCUSSION THREAD

**Discussion by klmcguire** [prody/ProDy#2081] [Archived Issues]
I tried targeted MD yesterday using CoMD.  I used the plugin for VMD (vmd v. 1.9.4a55 and newest v. of comd, namd v. 2.13 multicore cuda). I fill in the boxes of the plugin and run the tcl file on a linux cluster.  The job runs through many cycles of TMD and outputs the  walker1_trajectory.dcd and walker2_trajectory.dcd.  I use those in the analysis as described in the tutorial.  When I get to the PCA part of the tutorial, it crashes at this line: `pca.buildCovariance(combined_traj)`.  I get - ValueError: coordsets must have more than 3 coordinate sets.  What is the usual cause for this? I don't get any errors during the TMD simulation.  Is it giving me all of the outputs I need? Is it outputting them correctly? I read another post that mentioned needing more than one frame for the analysis. Is there a bug in the current version that overwrites the dcd with the current TMD cycle instead of adding to it? Images and TCL file provided below for more context. Thanks!

One more note:  After `dcd1 = Trajectory('walker1_trajectory.dcd')`, and calling dcd1, I get `<Trajectory: walker1_trajectory (next 0 of 1 frames; 129,031 atoms)>`.  Shouldn't there be more than 1 frame in the final dcd? Same with dcd2.

<img width="1489" alt="COMD_Error" src="https://github.com/prody/ProDy/assets/105521635/4c2ad963-57eb-42bb-836e-651bb18a7911">

[TMD.tcl.txt](https://github.com/prody/ProDy/files/14576660/TMD.tcl.txt)

**Comment by jamesmkrieger** [comment]
It is strange for sure that each dcd file only has 1 coordinate set if you ran standard CoMD. It should add one coordinate set to the final dcd file for each cycle. 

However, it looks like you’ve only got output for one cycle in walker1_pro and walker2_pro. Are you sure you’re using vmd to interpret the tcl file properly on the cluster?

**Comment by klmcguire** [comment]
> It is strange for sure that each dcd file only has 1 coordinate set if you ran standard CoMD. It should add one coordinate set to the final dcd file for each cycle.
> 
> However, it looks like you’ve only got output for one cycle in walker1_pro and walker2_pro. Are you sure you’re using vmd to interpret the tcl file properly on the cluster?

I run the TCL script on a gpu node using vmd -dispdev text -e TMD.tcl. While it's running, in the console it does keep saying "Now running TMD 1" even when it goes to a new cycle. It never says Now running TMD 2 or 3 or 4 etc. I'm not sure if it's a TCL script error or a VMD version problem.

**Comment by jamesmkrieger** [comment]
Ok, that looks like it should be right then. It’s possible there’s a problem with the vmd version but I don’t think so if it’s reporting like that.

**Comment by jamesmkrieger** [comment]
The tcl script looks right too and it should be copying and concatenating the frames right. 

Is it saying finished TMD 1 or 2 etc and then running minimisation 1 or 2 etc then finished minimisation? It should do a minimisation after each TMD before the next round of ANM NMA and concatenate the frames from the minimisations

**Comment by klmcguire** [comment]
> The tcl script looks right too and it should be copying and concatenating the frames right.
> 
> Is it saying finished TMD 1 or 2 etc and then running minimisation 1 or 2 etc then finished minimisation? It should do a minimisation after each TMD before the next round of ANM NMA and concatenate the frames from the minimisations

I don't ever see that happen.  It will run the Monte-Carlo and then go to TMD 1. TMD 1 runs for about 1.5 mins and then goes to another Monte-Carlo, and then TMD 1 again.  It will do that many times, and then finishes and gives me the walker dcd files.

**Comment by jamesmkrieger** [comment]
Ok, then that’s probably an issue with reading the file that triggers it to hit continue to the next loop without doing those steps. Hopefully we can have a look at it sometime in the next few weeks.

**Comment by jamesmkrieger** [comment]
We also have a development version of comd and of a hybrid simulation extension of adaptive anm inside prody within a branch that hasn’t been merged yet. See https://github.com/prody/ProDy/pull/1309

Feel free to test these too and let us know how they work. You’d have to clone jamesmkrieger/prody or add it as a remote and then fetch and checkout the hybrid branch and do a developer install of prody.

let us know if you need any help with that

**Comment by klmcguire** [comment]
> We also have a development version of comd and of a hybrid simulation extension of adaptive anm inside prody within a branch that hasn’t been merged yet. See #1309
> 
> Feel free to test these too and let us know how they work. You’d have to clone jamesmkrieger/prody or add it as a remote and then fetch and checkout the hybrid branch and do a developer install of prody.
> 
> let us know if you need any help with that

Ok, great I will look into that soon. Anything I can do to help make ProDy optimized. It's a great setup.

**Comment by jamesmkrieger** [comment]
Thank you!

**Comment by klmcguire** [comment]
> Thank you!

I have one question since you are familiar with what I am trying to do.  The motion in the open to close state is very small as you saw in the RMSD plots from the other thread. You recommended TMD for that instead of AANM.  I used the default settings in the comd plugin.  Perhaps the change from one conformation to the other is so quick that it happens in one frame?  For a small shift between conformations, what settings would you try to get more frames? Would I decrease the force used from the default 20,000 or maybe increase the TMD time from the default 10 ps? Thanks!

**Comment by jamesmkrieger** [comment]
I meant regular TMD not the one from CoMD, but maybe it does good enough. You can look at the dcd files in walker1_pro and walker2_pro and see what they show.

You could also try calculating RMSDs on your combined trajectory and both of those without PCA setting the initial and final structures with setCoords.
